onward/app/models/MostPopularNx2.scala

```diff
@@ -56,6 +56,21 @@ object OnwardItemNx2 {
     }
 
   }
+
+  // We ideally want this to be replaced by something else in the near future. Probably
+  // image-rendering or similar. But this will do for now.
+  // TODO: Replace this.
+
+  def getImageSources(imageMedia: Option[ImageMedia]): Map[String, Option[String]] = {
+    var images = for {
+      profile: ImageProfile <- List(Item300, Item460)
+      width: Int <- profile.width
+      trailPicture: ImageMedia <- imageMedia
+    } yield{
+      width.toString -> profile.bestSrcFor(trailPicture)
```

Comment:
I haven't really seen this pattern - what is it doing?

Reply:
We get the `ImageProfile` width (e.g. 300, 460 etc.,) which is technically Optional but always present for the specific `ImageProfile`s we would use. We then use that as the key in a `Map` to the image URL. 

It _should_ mean that, as compared to the list I originally used, we can grab specific images without worrying about the ordering. We can add new widths in here and it won't break the data model for DCR

Reply:
I'm in Scala land so pinch of salt for this one but I think this operation, where we choose the image based on width, should happen on the client. If I look at other places where this _is_ happening server side it seems to be for when we need to pick 1 url, like a thumbnail.

If I follow along correctly, then we want to be sending the `source`s to the client now (I think as `ImageMedia[]`). Previously we picked one and just used a url, but now we want to send the array so that we can create the sources using the image urls that get generated from Fastly.

Reply:
@oliverlloyd I think we are picking sources here. I.e. CAPI might provide a lot of source images for an image, and it doesn't necessarily make sense to send them all to DCR. But you are right that we could send more if we think it necessary.